core/state-db/src/pruning.rs

```diff
@@ -166,7 +166,7 @@ impl<BlockHash: Hash, Key: Hash> RefWindow<BlockHash, Key> {
 			inserted,
 			deleted,
 		};
-		let block = self.pending_number + self.window_size() as u64;
+		let block = self.pending_number + (self.death_rows.len() + self.pending_records.len()) as u64;
```

Comment:
Since pending prunings are always applied last they should not be accounted for here.

Reply:
Perhaps a comment in the code if it's an east conceptual mistake?